subsys/bluetooth/host/conn.c

```diff
@@ -2061,6 +2061,7 @@ struct bt_conn *bt_conn_create_slave_le(const bt_addr_le_t *peer,
 		return NULL;
 	}
 
+	conn->id = param->id;
```

Comment:
I think this should be set by `bt_conn_add_le(id, peer)` (we should add `id` as an argument to this function). Other places where this function seems to be connected with central role but if we don't extend the API now somebody can create a bug in the future.

I am not the maintainer of this stack so I don't want to impose my opinion here so no red x :)

Reply:
@pdunaj agreed, however that seems like an independent change, it's essentially refactoring which affects quite a lot of code, whereas the current patch in this PR is fixing the missing initialisation of conn->id. I wouldn't mind adding the refactoring patch to this PR though.

Reply:
Good idea. But in this case if you are fine we can merge this one first to have bug fixed. Then refactor can be done in separate PR.

Reply:
@pdunaj sounds like a plan